### PR TITLE
[stable/kubernetes-external-secrets]: add kubernetes-external-secrets chart

### PR DESCRIPTION
**Pull request by keweilu**
#### What this PR does / why we need it:

This chart deploys [kubernetes-external-secrets](https://github.com/godaddy/kubernetes-external-secrets) which allows you to use external secret management systems (e.g., AWS Secrets Manager) to securely add secrets in Kubernetes.

cc @silasbw @jeffpearce @JacopoDaeli

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by bilby91** [comment]
Looking forward into this one :)

File: stable/kubernetes-external-secrets/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+name: kubernetes-external-secrets
+version: 1.0.1
+appVersion: 1.2.0
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/Chart.yaml]: App version seems to be 1.3.0 as thats the image tag thats used in the values :)

File: stable/kubernetes-external-secrets/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{{- if .Values.rbac.create -}}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "kubernetes-external-secrets.fullname" . }}
+  labels:
+    app: {{ template "kubernetes-external-secrets.name" . }}
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/templates/rbac.yaml]: Chart is mixing old style labels with new style labels
```
labels:
    app.kubernetes.io/name: {{ include "kubernetes-external-secrets.name" . }}
    helm.sh/chart: {{ include "kubernetes-external-secrets.chart" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
```

File: stable/kubernetes-external-secrets/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+description: Kubernetes External Secrets CustomResourceDefinition
+keywords:
+  - kubernetes-external-secrets
+  - secrets
+home: https://github.com/godaddy/kubernetes-external-secrets
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/Chart.yaml:9]: add sources as well
```
sources:
  - https://github.com/godaddy/kubernetes-external-secrets
```

**Review by Flydiverny — CHANGES_REQUESTED** [review]
You probably want to add a values file for ci (in ci folder, [example](https://github.com/helm/charts/blob/master/stable/ambassador/ci/ci-values.yaml))

You also want to add an OWNERS ([example](https://github.com/helm/charts/blob/master/stable/ambassador/OWNERS)) file so you can maintain the chart :)

File: stable/kubernetes-external-secrets/Chart.yaml (hunk 1 above)
Comment (keweilu) [reply stable/kubernetes-external-secrets/Chart.yaml]: fixed, thanks for pointing this out

**Comment by Flydiverny** [comment]
/ok-to-test
Think I should be able to do this command 😅

File: stable/kubernetes-external-secrets/README.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+| `env.AWS_REGION`                     | Set AWS_REGION in Deployment Pod                             | `us-west-2`                                             |
+| `env.EVENTS_INTERVAL_MILLISECONDS`   | Set EVENTS_INTERVAL_MILLISECONDS in Deployment Pod           | `60000`                                                 |
+| `env.POLLER_INTERVAL_MILLISECONDS`   | Set POLLER_INTERVAL_MILLISECONDS in Deployment Pod           | `10000`                                                 |
+| `envVarsFromSecret.AWS_ACCESS_KEY_ID`     | Set AWS_ACCESS_KEY_ID (from a secret) in Deployment Pod      |                                                         |
+| `envVarsFromSecret.AWS_SECRET_ACCESS_KEY` | Set AWS_SECRET_ACCESS_KEY (from a secret) in Deployment Pod  |                                                         |
+| `image.repository`                   | kubernetes-external-secrets Image name                       | `godaddy/kubernetes-external-secrets`                   |
+| `image.tag`                          | kubernetes-external-secrets Image tag                        | `1.2.0`                                                 |
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/README.md]: 1.2.0 -> 1.3.1 :)

File: stable/kubernetes-external-secrets/README.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the `kubernetes-external-secrets` chart and their default values.
+
+| Parameter                            | Description                                                  | Default                                                 |
+| ------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------- |
+| `env.AWS_REGION`                     | Set AWS_REGION in Deployment Pod                             | `us-west-2`                                             |
+| `env.EVENTS_INTERVAL_MILLISECONDS`   | Set EVENTS_INTERVAL_MILLISECONDS in Deployment Pod           | `60000`                                                 |
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/README.md]: This doesn't exist in 1.3.1

File: stable/kubernetes-external-secrets/values.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+# Default values for kubernetes-external-secrets.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# Environment variables to set on deployment pod
+env:
+  AWS_REGION: us-west-2
+  EVENTS_INTERVAL_MILLISECONDS: 60000
```
Comment (Flydiverny) [inline stable/kubernetes-external-secrets/values.yaml]: removed in 1.3.1

**Review by Flydiverny — COMMENTED** [review]
Now I'm just nitpicking  :)

**Comment by keweilu** [comment]
@mattfarina and @sameersbn anything else we should do to get this merged?

**Comment by Flydiverny** [comment]
@v-yarotsky There's a chart available on helm hub already! :) 

https://hub.helm.sh/charts/external-secrets/kubernetes-external-secrets
its hosted from the https://github.com/godaddy/kubernetes-external-secrets/tree/master/charts/kubernetes-external-secrets repo

**Pull request closed without merging**